test/com/amazon/ion/streaming/ReaderTest.java

```diff
@@ -246,10 +248,12 @@ public void testReadingDecimalAsLong()
     @Test
     public void testReadingDecimalAsBigInteger()
     {
-        assertEquals(readBigInteger("0."), readBigInteger("0"));
-        assertEquals(readBigInteger("null.decimal"), readBigInteger("null.int"));
-        assertEquals(readBigInteger("9223372036854775807."), readBigInteger("9223372036854775807")); // Max long
-        assertEquals(readBigInteger("2d24"), readBigInteger("2000000000000000000000000"));
+        try {
+            readBigInteger("0.");
+            Assert.fail();
+        } catch (Exception e) {
+            // Pass
+        }
```

Comment:
```suggestion
        assertThrows(IllegalStateException.class, () -> readBigInteger("0."));
```

With the attendant import up above:
```java
import static org.junit.jupiter.api.Assertions.assertThrows;
```

Reply:
Sorry, I saw your approval and merged before I saw this comment. This would be a good change.

Reply:
No worries, I tossed this same comment on #522 which is not yet merged.